File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {

        unordered_set<int> hash(nums.begin(), nums.end());
        if (nums.size() == 0) return 0;
        int res = 1;

        for (int i = 0; i < nums.size(); i++) {
            if (hash.find(nums[i] - 1) != hash.end()) {
                int length = 2;
                while (hash.find(nums[i] - length) != hash.end()) {
                    length++;
                }
                res = max(res, length);
            }
        }

        return res;
    }
};

File: Data Structures & Algorithms/maximum-product-subarray/submission-2.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        res = nums[0]
        currMax, currMin = 1, 1

        for n in nums:
            tmp = currMax * n
            currMax = max(currMax * n, currMin * n, n)
            currMin = min(tmp, currMin * n, n)
            res = max(res, currMax)
        return res

        


# [1,2,-3,4,-2]
# currMax = 1, currMin = 1, res = 1
# currMax = 2, currMin = 1, res = 2
# currMax = -3, currMin = -6, res = 2
# currMax = 4, currMin = -24, res = 4
# currMax = 48, currMin = -8, res =48